app/code/Magento/Catalog/Observer/SetSpecialPriceStartDate.php

```diff
@@ -38,7 +38,8 @@ public function execute(\Magento\Framework\Event\Observer $observer)
     {
         /** @var  $product \Magento\Catalog\Model\Product */
         $product = $observer->getEvent()->getProduct();
-        if ($product->getSpecialPrice() && ! $product->getSpecialFromDate()) {
+        $specialFromDate = $product->getSpecialFromDate();
+        if ($product->getSpecialPrice() && !is_bool($specialFromDate) && !$specialFromDate) {
```

Comment:
Looks like it should be check for `=== null` instead. Also, cover corresponding case with unit test.

Reply:
@orlangur if i am using condition `if ($product->getSpecialPrice() && $specialFromDate === null)` it can save blank value as well.

Reply:
@orlangur , please update if it is correct.

Reply:
`is_bool` is very odd condition here. Should we do `$product->setData('special_from_date'` only when value is `null`?